internal/resolution/client/client.go

```diff
@@ -35,7 +35,7 @@ type DependencyClient interface {
 }
 
 type Registry struct {
-	URL string
+	EcosystemSpecific any
 }
```

Comment:
Might as well make this `type Registry any`/`interface{}` or remove the type altogether.